cccatalog-api/cccatalog/api/utils/watermark.py

```diff
@@ -1,15 +1,169 @@
-import requests
-import piexif
+import os
+from enum import Flag, auto
 from io import BytesIO
+from textwrap import wrap
+
+import piexif
+import requests
 from PIL import Image, ImageFont, ImageDraw
 
-horizontal_margin = 100
-vertical_margin = 200
-frame_color = '#fff'
-text_color = '#000'
+BREAKPOINT_DIMENSION = 400  # 400px
+MARGIN_RATIO = 0.04  # 4%
+FONT_RATIO = 0.04  # 4%
+
+FRAME_COLOR = '#fff'  # White frame
+TEXT_COLOR = '#000'  # Black text
+
+
+class Dimension(Flag):
+    """
+    This enum represents the two dimensions of an image
+    """
+
+    HEIGHT = auto()
+    WIDTH = auto()
+    BOTH = HEIGHT | WIDTH
+    NONE = 0
+
+
+# Utils
+
+def _smaller_dimension(width, height):
+    """
+    Determine which image dimensions are below the breakpoint dimensions
+    :param width: the width of the image
+    :param height: the height of the image
+    :return: True if the image is small, False otherwise
+    """
+
+    smaller_dimension = Dimension.NONE
+    if width < BREAKPOINT_DIMENSION:
+        smaller_dimension = smaller_dimension | Dimension.WIDTH
+    if height < BREAKPOINT_DIMENSION:
+        smaller_dimension = smaller_dimension | Dimension.HEIGHT
+    return smaller_dimension
+
+
+def _get_font_path(monospace=False):
+    """
+    Return the path to the TTF font file
+    :param monospace: True for monospaced font, False for variable-width font
+    :return: the path to the TTF font file
+    """
+
+    font_name = 'SourceSansPro-Bold.ttf' \
+        if monospace \
+        else 'SourceSansPro-Bold.ttf'
```

Comment:
Can you remove the `if else`? The values are the same

Reply:
That's a typo, it was supposed to be 
```
'SourceCodePro-Bold.ttf' if monospace else 'SourceSansPro-Bold.ttf'
```
I'll correct it immediately!